litedram/core/multiplexer.py

```diff
@@ -136,11 +136,11 @@ def __init__(self, txxd):
         # # #
 
         if txxd is not None:
-            count = Signal(max=max(txxd+1, 2))
+            count = Signal(max=max(txxd, 2))
             self.sync += \
                 If(valid,
                     count.eq(txxd),
-                    If((txxd - 1) == 0,
+                    If((txxd) == 0,
```

Comment:
For me it's doing the opposite, it wait an extra cycle. (load a > value), probably related to the other point.

We need to add unit-tests for these small modules, i'll try to look at that.

Reply:
It was supposed to change the line above... it appeared to work because of the first bug.  I'm testing a proper fix now.